gateway/src/apicast/configuration.lua

```diff
@@ -140,20 +142,48 @@ function _M.services_limit()
 end
 
 function _M.filter_services(services, subset)
-  subset = subset and util.to_hash(subset) or _M.services_limit()
-  if not subset or not next(subset) then return services end
-
   local s = {}
+  local serviceRegexpFilter  = env.value("APICAST_SERVICES_FILTER_BY_URL")
+
+  if serviceRegexpFilter then
+    -- Checking that the regexp sent is correct, if not an empty service list
+    -- will be returned.
+    local _, err = match("", serviceRegexpFilter)
+    if err then
+      -- @todo this return and empty list, Apicast will continue running maybe
+      -- process need to be stopped here.
+      ngx.log(ngx.ERR, "APICAST_SERVICES_FILTER cannot compile and all services are filtering out")
+      return s
```

Comment:
I know it was like this before the PR, but given that it's a variable used in the new code, maybe we should take this opportunity to give it a more descriptive name. `selected_services` or something similar.